app.py

```diff
@@ -140,62 +173,37 @@ def is_valid_dates(df, column):
         if not is_valid_dates(df, date_column):
             raise ValueError(f"Date column must be in YYYY-MM-DD format (e.g. 2023-12-31). Found values like: {df[date_column].iloc[0]} with dtype: {df[date_column].dtype}")
 
-
-        # Define and fit the MMM model
-        # import ipdb; ipdb.set_trace()
-
         logging.debug("Creating MMM model")
         mmm = MMM(
-            adstock=GeometricAdstock(l_max=int(adstock_max_lag)),
+            adstock=GeometricAdstock(l_max=adstock_max_lag),
             saturation=LogisticSaturation(),
             date_column=date_column,
             channel_columns=channel_columns,
             control_columns=control_columns,
             yearly_seasonality=yearly_seasonality,
         )
         logging.info("MMM model defined.")
-        
-        X = df.drop('sales', axis=1)
-        y = df['sales']
-
-        logging.debug("Starting model fitting.")
 
-        # mmm.fit(X, y, random_seed=42, cores=1)
-        mmm.fit(X, y)
-        logging.info("Model fitting completed.")
-        
-        # Extract and return summary statistics
-        summary = az.summary(mmm.fit_result)
-        
-        # Filter only the most important statistics
-        important_params = summary[summary.index.str.contains('alpha|beta|sigma|intercept|lam|gamma_control', case=False)]
-        # Limit decimal places and convert to more compact format
-        important_params = important_params.round(5)
+        # Ensure date_week is in datetime format
+        df[date_column] = pd.to_datetime(df[date_column])
         
-        summary_json = important_params.to_json(orient="split", double_precision=5)
-        logging.info("Summary statistics extracted.")
-        logging.info("summary_json=%s", summary_json)
         
-        # Add model metrics
-        response = {
-            "status": "completed",
-            "summary": summary_json,
-            # "model_info": {
-            #     "num_observations": len(df),
-            #     "channels": channel_columns,
-            #     "adstock_max_lag": adstock_max_lag,
-            #     "yearly_seasonality": yearly_seasonality
-            # }
-        }
-
+        # X = df.drop(y_column, axis=1).astype(float)
+        X = df.drop(y_column, axis=1)
+        y = df[y_column].astype(float)
+            
+        mmm.fit(X, y)
+        logging.info("Model fitting completed.")
         logging.info("run_mmm_task completed successfully.")
-        logging.debug("response=%s", response)
 
-        return response
+        logging.info("Serializing MMM model")
+        mmm_pickle = pickle.dumps(mmm)
+        logging.info("MMM model serialized successfully")
+        return mmm_pickle
```

Comment:
are you sending the pickled model object to the server?

Reply:
Celery requires that an object is serializable so it can be sent to and from the celery worker for processing. The MMM instance is not serializable with pickle (I believe there are functions that pickle can't handle) so I'm using dill which is more robust to arbitrary function definitions. Looking into it, there is a cleaner way to do this which I'll implement

Reply:
See [154f57a](https://github.com/pymc-labs/gpt-bayes/pull/8/commits/154f57a6f709e128a2553f7f5e976e6f008e3f2e) where I registered a 'custom' serializer which is just dill

Reply:
@twiecki are you good with this?